### PR TITLE
Moving all the SOW Builder files into the feature branch - 719

### PR DESCRIPTION
**Pull request by nagmak**
### What changes were made?

- Bounty context to handle all the bounty data across nested parent and children components
- Being able to edit and save data on Project modals
- Backend function to handle submitting a bounty

### Is there any background info about the change?

- Part of the SOW Builder

### Are there any state changes? Mention key ones (if any)

Quite a lot... But essentially look at the Bounty context for the major ones

### Screenshots or Gifs✨
Screen Shot 2022-06-09 at 1 46 18 PM

### Any new dependencies? Why were they added?

NA

### Shortcut Link

NA

### CODE REVIEW

File: components/Marketplace/BountyList.tsx [hunk 1]
```diff
@@ -46,6 +46,7 @@ export const BountyList = ({ contracts, error, sortParams, compare }) => {
       overflowY={"scroll"}
       scrollBehavior={"smooth"}
     >
+      {console.log(contracts)}
```
Comment (hone1er) [inline components/Marketplace/BountyList.tsx:49]: Left a console.log in there

File: components/Project/ProjectDescription.tsx [hunk 2]
```diff
@@ -39,7 +41,7 @@ const ProjectDescription = (props) => {
               src={"twali-assets/editicon.png"}
             />
           </Button>
```
Comment (hone1er) [inline components/Project/ProjectDescription.tsx:43]: This closing tag can be removed. The <Button/> above is self closing

File: components/Project/ProjectHeader.tsx [hunk 3]
```diff
@@ -1,11 +1,30 @@
...
+    ...userState,
+    // userName: "",
+    // userWallet: "",
+    // uuid: "",
+  });
+  const { setBounty, ...bountyState} = useBounty();
+  const [bountyData, setBountyData] = useState<Bounty>({
+    ...bountyState,
+    contractOwnerUserName: userData.userName,
+    contractCreatedOn: 1651968000,
```
Comment (hone1er) [inline components/Project/ProjectHeader.tsx:23]: Why was this createdOn time chosen??

File: components/Project/ProjectDescription.tsx (hunk 2 above)
Comment (nagmak) [reply components/Project/ProjectDescription.tsx:43]: oh i that's part of the button! the image is inside it so it's needed

`              <Button
            onClick={onProjectDescModalOpen}
            alignSelf="flex-end"
            variant="ghost"
            aria-label="Update project title"
            margin={"0px"}
            mx={"0px"}
            pos={"relative"}
            width={"fit-content"}
            borderRadius={"md"}
          >
            
          </Button>`

File: components/Project/ProjectHeader.tsx (hunk 3 above)
Comment (nagmak) [reply components/Project/ProjectHeader.tsx:23]: this is gonna change, we should be getting the contract creation time from the solidity smart contract whenever it gets generated... so I kept it a constant value until they're fully hooked up

File: pages/_app.tsx [hunk 4]
```diff
@@ -32,7 +33,9 @@ function App({ Component, pageProps }: AppProps) {
     <UserProvider>
       <ChakraProvider theme={twaliTheme}>
       <TokenContext>
```
Comment (hone1er) [inline pages/_app.tsx:35]: Last thing and it's good to go. The TokenContext here needs to be renamed to TokenProvider

Comment (nagmak) [reply pages/_app.tsx]: doneee!

**Review by hone1er — APPROVED** [review]
Looking good 👍

**Pull request merged**